### PR TITLE
Expose resolved .whl files in filegroup target of pypi__ packages.

### PR DESCRIPTION
**Pull request by hrfuller**
This allows distribution tools that have their own mechanisms of
packaging whls to take advantage of the pip resolve and dependency
graph creation that rules_python_external offers.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Does not include precompiled binaries, eg. `.par` files. See [CONTRIBUTING.md](/CONTRIBUTING.md) for info
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
The downloaded .whl files used to create py_library targets in the repository rule are deleted after the contents is extracted.

Issue Number: N/A

## What is the new behavior?
The .whl files are copied to the new //pypi__ package root and a filegroup target called `:whl` is created to capture the .whl file and its direct wheel dependencies. A dependency graph of wheel files is created in parallel to the py_library dependency graph.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by thundergolfer** [comment]
On first scan this looks good. Will look to revie and test out post-https://github.com/bazelbuild/rules_python/pull/359.

**Comment by hrfuller** [comment]
Hey @thundergolfer since #359  landed, can you PTAL when you have a chance?

File: python/pip_install/extract_wheels/lib/bazel_test.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            enable_implicit_namespace_pkgs=False,
+        )[2:]  # Take off the leading // from the returned label.
+        # Assert that the raw wheel ends up in the package.
+        self.assertIn(wheel_name, os.listdir(generated_bazel_dir))
+        with open("{}/BUILD".format(generated_bazel_dir)) as build_file:
+            build_file_content = build_file.read()
+            self.assertIn('filegroup', build_file_content)
+
+
+
```
Comment (thundergolfer) [inline python/pip_install/extract_wheels/lib/bazel_test.py:25]: nit: one too many newlines here.

File: python/pip_install/README.md [hunk 2]
```diff
@@ -64,18 +64,21 @@ pip_install(
...
+load("@py_deps//:requirements.bzl", "requirement", "whl_requirement")
 
 py_binary(
     name = "main",
     srcs = ["main.py"],
     deps = [
         requirement("boto3"),
     ],
+    data = [
+        whl_requirement("boto3"),
```
Comment (thundergolfer) [inline python/pip_install/README.md:76]: nit: Think this needs a comment to communicate why you'd do this, as this particular target will run just fine without the addition, so I expect the example won't be all that clear to end-users.

**Review by thundergolfer — APPROVED** [review]

File: python/pip_install/extract_wheels/lib/BUILD [hunk 3]
```diff
@@ -41,6 +41,19 @@ py_test(
     ],
 )
 
+py_test(
+    name = "bazel_test",
```
Comment (thundergolfer) [inline python/pip_install/extract_wheels/lib/BUILD:45]: nit: Think this test name (and associated file) could be more specific given it's only testing the new wheel filegroup. `whl_filegroup_test`?

Comment (hrfuller) [reply python/pip_install/extract_wheels/lib/BUILD:45]: Yeah I get that, the idea was the name should be more general to allow expanding to include other tests of the bazel.py module. Happy to change it for the moment.

**Comment by hrfuller** [comment]
@thundergolfer This is ready to merge.

**Pull request merged**